R/rectangle.R

```diff
@@ -393,7 +396,12 @@ simplify_col <- function(x, nm, ptype = list(), transform = list(), simplify = F
     }
   }
 
-  n <- map_int(x, vec_size)
+  if (vec_is_list(x)) {
+    n <- list_sizes(x)
+  } else {
+    n <- map_int(x, vec_size)
```

Comment:
I don't quite follow the logic here — if `x` is not a list then I'm guessing it's either an atomic vector or a data frame? So why do we need the map?